sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs

```diff
@@ -40,6 +30,7 @@ protected async Task PreProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
             // if this policy doesn't have _scope cached try to get it from the static challenge cache.
             if (_scope != null)
             {
+                await SetAuthorizationHeader(message, new TokenRequestContext(_scope.Scopes, message.Request.ClientRequestId), async).ConfigureAwait(false);
```

Comment:
You can merge the _scope != null and the _scopeCache.TryGetValue condition to simplify the code a bit.